### PR TITLE
Static Linking, Premake, Toshi::TApplication, Toshi::TLog, TClass Registration, Precompiled Header, README, spdlog

### PR DESCRIPTION
**Pull request by InfiniteC0re**

### CODE REVIEW

File: Toshi/Source/Toshi/TClass.cpp [hunk 1]
```diff
@@ -61,14 +37,14 @@ bool __thiscall Toshi::TClass::AttachClassToParent()
 	return false;
 }
 
-Toshi::TClass::TClass(const char* name, TClass* parent, t_CreateTObject create, t_CreateTObjectAtPlace createAtPlace, t_RegisterScriptingAPI scripting, uint32_t version, size_t size)
+Toshi::TClass::TClass(const char* name, TClass* parent, t_CreateTObject create, t_CreateTObjectAtPlace createAtPlace, t_RegisterScriptingAPI scripting, uint16_t verMajor, uint16_t verMinor, size_t size)
```
Comment (AdventureT) [inline Toshi/Source/Toshi/TClass.cpp]: That's ok but i want to keep it as JPOG and de blob does it, i think.

File: Toshi/Source/Toshi/TMemory.cpp [hunk 2]
```diff
@@ -11,7 +11,7 @@ namespace Toshi
 		TMemManager* TMemManager::s_Class;
 		size_t TMemManager::s_GlobalBlockSize;
 
-		void TMemManager::Initialise(size_t globalBlockSize)
+		void TMemManager::Initialize(size_t globalBlockSize)
```
Comment (AdventureT) [inline Toshi/Source/Toshi/TMemory.cpp]: Same here, i wanted to also keep that spelling mistake they did on JPOG

Comment (InfiniteC0re) [reply Toshi/Source/Toshi/TMemory.cpp]: oh I didn't even know there was a mistake in JPOG, it was mine mistake lol

File: Toshi/Source/Toshi/TClass.cpp (hunk 1 above)
Comment (InfiniteC0re) [reply Toshi/Source/Toshi/TClass.cpp]: but why? isn't this better? 🤔

**Pull request merged**